tests/test_nanopub.py

```diff
@@ -9,194 +9,211 @@
-    assert (None, RDF.type, namespaces.NP.Nanopublication) in nanopub.rdf
-    assert (None, namespaces.NP.hasAssertion, None) in nanopub.rdf
-    assert (None, namespaces.NP.hasProvenance, None) in nanopub.rdf
-    assert (None, namespaces.NP.hasPublicationInfo, None) in nanopub.rdf
-
-    new_concept = rdflib.term.URIRef('www.purl.org/new/concept/test')
-    nanopub = Nanopub.from_assertion(assertion_rdf, introduces_concept=new_concept)
-
-    assert nanopub.rdf is not None
-    assert (None, RDF.type, namespaces.NP.Nanopublication) in nanopub.rdf
-    assert (None, namespaces.NP.hasAssertion, None) in nanopub.rdf
-    assert (None, namespaces.NP.hasProvenance, None) in nanopub.rdf
-    assert (None, namespaces.NP.hasPublicationInfo, None) in nanopub.rdf
-
-    assert (None, namespaces.NPX.introduces, new_concept) in nanopub.rdf
-
-
-def test_nanopub_claim():
-    client = NanopubClient()
-    client.java_wrapper.publish = mock.MagicMock()
-    optional_triple = (rdflib.term.URIRef('http://www.uri1.com'),
-                       rdflib.term.URIRef('http://www.uri2.com'),
-                       rdflib.Literal('Something'))
-    client.claim('Some controversial statement', rdftriple=optional_triple)
-
-
-def test_nanopub_publish():
-    test_uri = 'http://www.example.com/my-nanopub'
-    test_concept_uri = 'http://purl.org/person#DrBob'  # This nanopub introduced DrBob
-    client = NanopubClient()
-    client.java_wrapper.publish = mock.MagicMock(return_value=test_uri)
-    assertion_rdf = rdflib.Graph()
-    assertion_rdf.add((namespaces.AUTHOR.DrBob, namespaces.HYCL.claims, rdflib.Literal('This is a test')))
-
-    nanopub = Nanopub.from_assertion(
-        assertion_rdf=assertion_rdf,
-        uri=rdflib.term.URIRef(test_uri),
-        introduces_concept=namespaces.AUTHOR.DrBob,
-        derived_from=rdflib.term.URIRef('http://www.example.com/another-nanopub')
-        )
-    pubinfo = client.publish(nanopub)
-    assert pubinfo['nanopub_uri'] == test_uri
-    assert pubinfo['concept_uri'] == test_concept_uri
-
-
-def test_nanopub_publish_blanknode():
-    test_concept = rdflib.term.BNode('test')
-    test_published_uri = 'http://www.example.com/my-nanopub'
-    expected_concept_uri = 'http://www.example.com/my-nanopub#test'
-    client = NanopubClient()
-    client.java_wrapper.publish = mock.MagicMock(return_value=test_published_uri)
-    assertion_rdf = rdflib.Graph()
-    assertion_rdf.add(
-        (test_concept, namespaces.HYCL.claims, rdflib.Literal('This is a test')))
-
-    nanopub = Nanopub.from_assertion(
-        assertion_rdf=assertion_rdf,
-        introduces_concept=test_concept,
-    )
-    pubinfo = client.publish(nanopub)
-    assert pubinfo['nanopub_uri'] == test_published_uri
-    assert pubinfo['concept_uri'] == expected_concept_uri
+        with mock.patch('nanopub.nanopub.profile', _get_mock_profile()):
+            result = Nanopub.from_assertion(assertion_rdf=assertion)
+
+            prov_entries = list(
```

Comment:
I would prefer something like `assert triple in result.rdf`, which would be easier to read.
Also, you could reuse TEST_ORCID.

Reply:
Yeah that would be more readable. The problem is that I do not want to assume what the subject in the triple would look like.
I know it looks like `(SOMETHING, prov:wasAttributedTo, <https://orcid.org/pietje>)`
And from running the test with the debugger I know SOMETHING to be `http://purl.org/nanopub/temp/mynanopub#`

Is it reasonable to assume that it will always be that? Right now the test only checks the values I'm interested in.

About reusing `TEST_ORCID` I am never sure if I should reuse a value that I have also used as input so that is why I copied it. However, I'm never consistent with this is I will just replace it with the `TEST_ORCID`

Reply:
See [rdflib documentation](https://rdflib.readthedocs.io/en/stable/intro_to_graphs.html):

Note that this triple does not have to be completely bound:
```python
if (bob, None, None) in graph:
    print("This graph contains triples about Bob!")
```

I think you should expect it to be the value in TEST_ORCID, not caring about what that value is. That makes it also clearer to read (you don't need to worry about whether something actually changed in the string)

Reply:
Nice I changed it

Reply:
Great